### PR TITLE
Add DroneCAN and CAN Stats sent on the wire

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: Tools/AP_Periph/can.cpp [hunk 1]
```diff
@@ -1208,16 +1212,60 @@ static void processTx(void)
...
             } else {
+                protocol_stats.tx_errors++;
                 canardPopTxQueue(&dronecan.canard);
             }
             break;
         }
     }
 }
 
+static void update_rx_protocol_stats(int16_t res)
```
Comment (tridge) [inline Tools/AP_Periph/can.cpp:1226]: need to default this off and have a check on board flash size >= 256k

File: libraries/AP_DroneCAN/AP_DroneCAN.cpp [hunk 2]
```diff
@@ -481,6 +487,42 @@ void AP_DroneCAN::send_node_status(void)
     _node_status_last_send_ms = now;
     node_status_msg.uptime_sec = now / 1000;
     node_status.broadcast(node_status_msg);
+
+    if (!option_is_set(Options::DISABLE_STATS)) {
```
Comment (tridge) [inline libraries/AP_DroneCAN/AP_DroneCAN.cpp:491]: i think it should be an ENABLE_STATS option

File: Tools/AP_Periph/can.cpp [hunk 3]
```diff
@@ -1208,16 +1212,60 @@ static void processTx(void)
         } else {
             // just exit and try again later. If we fail 8 times in a row
             // then start discarding to prevent the pool filling up
-            if (dronecan.tx_fail_count < 8) {
+            if (dronecan.tx_fail_count < 80) {
```
Comment (tridge) [inline Tools/AP_Periph/can.cpp:1215]: what is this?

**Review by tridge — CHANGES_REQUESTED** [review]

File: Tools/AP_Periph/Parameters.cpp [hunk 4]
```diff
@@ -551,6 +551,13 @@ const AP_Param::Info AP_Periph_FW::var_info[] = {
   #endif
 #endif
 
+    // @Param: CAN_STATS
+    // @DisplayName: CAN Statistics
+    // @Description: Enable CAN and Protocol statistics
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Advanced
+    GSCALAR(node_stats, "CAN_NODE_STATS", 1),
```
[the remark this answers is not in the document]
Comment (bugobliterator) [reply Tools/AP_Periph/Parameters.cpp]: changed to using DEBUG option bit

File: Tools/AP_Periph/can.cpp (hunk 1 above)
Comment (bugobliterator) [reply Tools/AP_Periph/can.cpp:1233]: Done

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
merging with flapping balance bot test

**Pull request merged**